packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -72,100 +59,23 @@ export function Comments( {
 		);
 	}, [] );
 
-	const CommentBoard = ( { thread, parentThread } ) => {
-		return (
-			<>
-				<CommentHeader
-					thread={ thread }
-					onResolve={ () => {
-						setActionState( {
-							action: 'resolve',
-							id: parentThread?.id ?? thread.id,
-						} );
-						setIsConfirmDialogOpen( true );
-					} }
-					onEdit={ () =>
-						setActionState( { action: 'edit', id: thread.id } )
-					}
-					onDelete={ () => {
-						setActionState( { action: 'delete', id: thread.id } );
-						setIsConfirmDialogOpen( true );
-					} }
-					onReply={
-						! parentThread
-							? () =>
-									setActionState( {
-										action: 'reply',
-										id: thread.id,
-									} )
-							: undefined
-					}
-					status={ parentThread?.status ?? thread.status }
-				/>
-				<HStack
-					alignment="left"
-					spacing="3"
-					justify="flex-start"
-					className="editor-collab-sidebar-panel__user-comment"
-				>
-					<VStack
-						spacing="3"
-						className="editor-collab-sidebar-panel__comment-field"
-					>
-						{ 'edit' === actionState?.action &&
-							thread.id === actionState?.id && (
-								<CommentForm
-									onSubmit={ ( value ) => {
-										onEditComment( thread.id, value );
-										setActionState( false );
-									} }
-									onCancel={ () => setActionState( false ) }
-									thread={ thread }
-									submitButtonText={ _x( 'Update', 'verb' ) }
-								/>
-							) }
-						{ ( ! actionState ||
-							'edit' !== actionState?.action ) && (
-							<RawHTML>{ thread?.content?.raw }</RawHTML>
-						) }
-					</VStack>
-				</HStack>
-				{ 'resolve' === actionState?.action &&
-					thread.id === actionState?.id && (
-						<ConfirmDialog
-							isOpen={ isConfirmDialogOpen }
-							onConfirm={ handleConfirmResolve }
-							onCancel={ handleCancelDelete }
-							confirmButtonText="Yes"
-							cancelButtonText="No"
-						>
-							{
-								// translators: message displayed when confirming an action
-								__(
-									'Are you sure you want to mark this comment as resolved?'
-								)
-							}
-						</ConfirmDialog>
-					) }
-				{ 'delete' === actionState?.action &&
-					thread.id === actionState?.id && (
-						<ConfirmDialog
-							isOpen={ isConfirmDialogOpen }
-							onConfirm={ handleConfirmDelete }
-							onCancel={ handleCancelDelete }
-							confirmButtonText="Yes"
-							cancelButtonText="No"
-						>
-							{
-								// translators: message displayed when confirming an action
-								__(
-									'Are you sure you want to delete this comment?'
-								)
-							}
-						</ConfirmDialog>
-					) }
-			</>
-		);
+	// Set active thread when block is selected.
+	useEffect( () => {
+		setActiveThread( blockCommentId ?? null );
+		setFocusThread( null );
+		setShowCommentBoard( false );
+	}, [ blockCommentId ] );
```

Comment:
Generally, setting state in an effect like this is not a good idea. Maybe just unmount the component with a `key` so the local state within is automatically reset?

Reply:
The `key` suggestion does not work?